.github/workflows/build-vault-oss.yml

```diff
@@ -0,0 +1,111 @@
+---
+name: build_vault
+
+# This workflow is intended to be called by the build workflow for each Vault
+# binary that needs to be built and packaged. The ci make targets that are
+# utilized automatically determine build metadata and handle building and
+# packing vault.
+
+on:
+  workflow_call:
+    inputs:
+      bundle-path:
+        required: false
+        type: string
+      cgo-enabled:
+        type: string
+        default: 0
+      create-packages:
+        type: boolean
+        default: true
+      goos:
+        required: true
+        type: string
+      goarch:
+        required: true
+        type: string
+      go-tags:
+        type: string
+      go-version:
+        type: string
+      package-name:
+        type: string
+        default: vault
+      vault-version:
+        type: string
+        required: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Vault ${{ inputs.goos }} ${{ inputs.goarch }} v${{ inputs.vault-version }}
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
Per security request, all actions need to be pinned to a specific sha (.github/workflows/actionlint.yml in this PR has done this correctly)

Reply:
Hi, thanks for the review @modrake! I've raised this PR accidentally earlier today and I just realised that. I will close this one, but will make sure to address your comments on the correct one (19770).